pkg/cmd/service/service_test.go

```diff
@@ -20,118 +25,158 @@ import (
+		NotBefore:             time.Now().Add(-time.Minute),
+		NotAfter:              time.Now().Add(time.Hour),
+		Subject: pkix.Name{
+			Country:      []string{"Earth"},
+			Organization: []string{"Starfleet"},
+		},
+		DNSNames: []string{"localhost"},
+	}
+
+	serverCert, serverPEM, err := genCert(template, caCert, &priv.PublicKey, caKey)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return serverCert, serverPEM, priv, nil
+}
+
+func writeToTempFile(data []byte) (*os.File, error) {
+	file, err := os.CreateTemp("", "openfga_tls_test")
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = file.Write(data)
+	if err != nil {
+		return nil, err
+	}
 
-	_, err = certFile.Write([]byte(serverCert))
-	require.NoError(t, err, "failed to write pem")
+	return file, nil
+}
 
-	keyFile, err := ioutil.TempFile("", "server_key")
-	require.NoError(t, err, "error creating pem file")
-	defer keyFile.Close()
+type certChain struct {
+	caCert         *x509.Certificate
+	serverCertFile string
+	serverKeyFile  string
+}
+
+func (c certChain) Clean() {
+	os.Remove(c.serverKeyFile)
+	os.Remove(c.serverKeyFile)
+}
 
-	_, err = keyFile.Write([]byte(serverKey))
-	require.NoError(t, err, "failed to write pem")
+// createKeys generates a self-signed root CA certificate and a server certificate and server key. It will write
+// the PEM encoded server certificate and server key to temporary files. It is the responsibility of the caller
+// to delete these files by calling `Clean` on the returned `certChain`.
+func createKeys(t *testing.T) certChain {
+	caCert, _, caKey, err := genCACert()
+	require.NoError(t, err)
+
+	_, serverPEM, serverKey, err := genServerCert(caCert, caKey)
+	require.NoError(t, err)
+
+	serverCertFile, err := writeToTempFile(serverPEM)
+	require.NoError(t, err)
+
+	serverKeyFile, err := writeToTempFile(pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: x509.MarshalPKCS1PrivateKey(serverKey),
+		},
+	))
+	require.NoError(t, err)
 
 	require.NoError(t, os.Setenv(httpTLSEnabledEnvVar, "true"), "failed to set env var")
-	require.NoError(t, os.Setenv(httpTLSCertPathEnvVar, certFile.Name()), "failed to set env var")
-	require.NoError(t, os.Setenv(httpTLSKeyPathEnvVar, keyFile.Name()), "failed to set env var")
+	require.NoError(t, os.Setenv(httpTLSCertPathEnvVar, serverCertFile.Name()), "failed to set env var")
+	require.NoError(t, os.Setenv(httpTLSKeyPathEnvVar, serverKeyFile.Name()), "failed to set env var")
 
 	require.NoError(t, os.Setenv(grpcTLSEnabledEnvVar, "true"), "failed to set env var")
-	require.NoError(t, os.Setenv(grpcTLSCertPathEnvVar, certFile.Name()), "failed to set env var")
-	require.NoError(t, os.Setenv(grpcTLSKeyPathEnvVar, keyFile.Name()), "failed to set env var")
+	require.NoError(t, os.Setenv(grpcTLSCertPathEnvVar, serverCertFile.Name()), "failed to set env var")
+	require.NoError(t, os.Setenv(grpcTLSKeyPathEnvVar, serverKeyFile.Name()), "failed to set env var")
 
-	return certFile.Name()
+	return certChain{
```

Comment:
I don't like this name, but I also can't think of a better one.

Reply:
Yeah, a cert chain is a specific thing. This is more like a `certHandle` or `certWrapped` or something like that 🤷 ? I usually like to use the term "handle" if a struct embeds/groups/aggregates other commonly used parameters.

Reply:
Thanks! Went with `certHandle`.